src/D2L.CodeStyle.Analyzers/Immutability/ImmutabilityAnalyzer.cs

```diff
@@ -228,6 +235,56 @@ SimpleNameSyntax syntax
 			ctx.ReportDiagnostic( diagnostic );
 		}
 
+		private static void AnalyzeConflictingImmutabilityOnMember(
+			SyntaxNodeAnalysisContext ctx ) {
+
+			// Ensure syntax is expected and get the symbol
+			if( ctx.Node is not TypeDeclarationSyntax syntax ) {
+				return;
+			}
+			var symbol = ctx.SemanticModel.GetDeclaredSymbol( syntax );
+
+			// Get information about immutability
+			bool hasImmutable = Attributes.Objects.Immutable.IsDefined( symbol );
+			bool hasConditionallyImmutable = Attributes.Objects.ConditionallyImmutable.IsDefined( symbol );
+			bool hasImmutableBase = Attributes.Objects.ImmutableBaseClass.IsDefined( symbol );
+
+			// Check if there are conflicting immutability attributes
+			if( hasImmutable && hasConditionallyImmutable ) {
+				// [Immutable] and [ConditionallyImmutable] both exist,
+				// so create a diagnostic
+				var diagnostic = Diagnostic.Create(
+					Diagnostics.ConflictingImmutability,
+					syntax.Identifier.GetLocation(),
+					"Immutable",
+					"ConditionallyImmutable",
+					syntax.Keyword );

```

Comment:
nit: these `);` should be on the next line (iirc there is no editorconfig for this yet)